sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -139,9 +140,8 @@ private[sql] case class InsertIntoHadoopFsRelation(
     }
 
     def writeRows(taskContext: TaskContext, iterator: Iterator[Row]): Unit = {
-      writerContainer.executorSideSetup(taskContext)
-
       try {
+        writerContainer.executorSideSetup(taskContext)
```

Comment:
Seems it is possible that `executorSideSetup` fails before setting up `outputCommitter`. Then, when we call `abortTask`, we will get a NPE since we are calling `outputCommitter.abortTask(taskAttemptContext)`.